src/providers.jl

```diff
@@ -1,17 +1,42 @@
+# Hardcoded list of supported platforms
+# In principle, we could check FFTW_jll.is_available() and MKL_jll.is_available()
+# but then we would have to load MKL_jll which we want to avoid (lazy artifacts!)
+const platforms_providers = Dict(
+    Base.BinaryPlatforms.Platform("aarch64", "macos") => ("fftw",),
+    Base.BinaryPlatforms.Platform("aarch64", "linux"; libc = "glibc") => ("fftw",),
+    Base.BinaryPlatforms.Platform("aarch64", "linux"; libc = "musl") => ("fftw",),
+    Base.BinaryPlatforms.Platform("armv6l", "linux"; libc = "glibc", call_abi = "eabihf") => ("fftw",),
+    Base.BinaryPlatforms.Platform("armv6l", "linux"; libc = "musl", call_abi = "eabihf") => ("fftw",),
+    Base.BinaryPlatforms.Platform("armv7l", "linux"; libc = "glibc", call_abi = "eabihf") => ("fftw",),
+    Base.BinaryPlatforms.Platform("armv7l", "linux"; libc = "musl", call_abi = "eabihf") => ("fftw",),
+    Base.BinaryPlatforms.Platform("i686", "linux"; libc = "glibc") => ("fftw", "mkl"),
+    Base.BinaryPlatforms.Platform("i686", "linux"; libc = "musl") => ("fftw",),
+    Base.BinaryPlatforms.Platform("i686", "windows") => ("fftw", "mkl"),
+    Base.BinaryPlatforms.Platform("powerpc64le", "linux"; libc = "glibc") => ("fftw",),
+    Base.BinaryPlatforms.Platform("x86_64", "macos") => ("fftw", "mkl"),
+    Base.BinaryPlatforms.Platform("x86_64", "linux"; libc = "glibc") => ("fftw",),
+    Base.BinaryPlatforms.Platform("x86_64", "linux"; libc = "musl") => ("fftw",),
+    Base.BinaryPlatforms.Platform("x86_64", "freebsd") => ("fftw",),
+    Base.BinaryPlatforms.Platform("x86_64", "windows") => ("fftw", "mkl"),
+)
+const valid_fftw_providers = Base.BinaryPlatforms.select_platform(platforms_providers, Base.BinaryPlatforms.HostPlatform())
```

Comment:
If you do something like
```julia
const valid_fftw_providers = let
    platforms_providers = Dict(...)
    Base.BinaryPlatforms.select_platform(platforms_providers, Base.BinaryPlatforms.HostPlatform())
end
```
you avoid storing in the compile cache a useless dictionary.